Src/DependencyCollector/Shared.Tests/Implementation/FrameworkHttpProcessingTest.cs

```diff
@@ -395,6 +395,50 @@ public void RddTestHttpProcessingFrameworkOnBeginSkipsAddingSourceHeaderPerConfi
             Assert.AreEqual(0, request.Headers.Keys.Cast<string>().Where((x) => { return x.StartsWith("x-ms-", StringComparison.OrdinalIgnoreCase); }).Count());
         }
 
+        [TestMethod]
+        [Description("Ensures that an empty header is added for the first request, but once the appId is retrieved, that's not the case anymore.")]
+        public void RddTestHttpProcessingFrameworkOnBeginAddsEmptyCorrelationHeaderForFirstCall()
+        {
+            var correlationIdLookupHelper = new CorrelationIdLookupHelper((ikey) =>
+            {
+                // Pretend App Id is the same as Ikey
+                var tcs = new TaskCompletionSource<string>();
+                tcs.SetResult(ikey + "-appId");
+                return tcs.Task;
+            });
+
+            this.httpProcessingFramework.OverrideCorrelationIdLookupHelper(correlationIdLookupHelper);
+
+            var request = WebRequest.Create(this.testUrl);
+            this.httpProcessingFramework.OnRequestSend(request);
+
+            // Header with empty app Id.
+            Assert.AreEqual(request.Headers[RequestResponseHeaders.RequestContextHeader], "appId=cid-v1:");
+
+            // Wait for the appId fetch to complete
+            while (correlationIdLookupHelper.IsFetchAppInProgress(this.configuration.InstrumentationKey))
+            {
+                Thread.Sleep(10); // wait 10 ms.
```

Comment:
potentially infinite. use TestTimeout for the Testcase to avoid.

Reply:
I added [Timeout(5000)]